### ISSUE TITLE
Click `a` element which has no `href` causes TypeError

### ISSUE THREAD

**Issue by leaysgur** [BuilderIO/partytown#189]
**Describe the bug**

When using `partytown` with GoogleTagManager, clicking `a` element without `href` attribute causes the error.

```
TypeError: Failed to construct 'URL': Invalid URL
```

At https://github.com/BuilderIO/partytown/blob/c124a7def5828f5f92b953dbddc5eb48d7d450c1/src/lib/web-worker/worker-anchor.ts#L20

**To Reproduce**

1. Setup repro(see below)
2. Open DevTools console
3. Click last item(3/3)

**Reproduction link**

Since we do not have sharable GTM id, I add complete HTML source as reproduction instead.

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <script
      type="text/partytown"
      src="https://www.googletagmanager.com/gtag/js?id={{⚠️REPLACE_YOUR_GTM_ID_HERE}}"
    ></script>
    <script type="text/partytown">
      window.dataLayer = window.dataLayer || [];
      function gtag() {
        dataLayer.push(arguments);
      }
      gtag("js", new Date());
      gtag("config", "{{⚠️REPLACE_YOUR_GTM_ID_HERE}}");
    </script>
    <script>!(function(w,p,f,c){c=w[p]=Object.assign(w[p]||{},{"lib":"/~partytown/","debug":false});c[f]=(c[f]||[]).concat(["dataLayer.push"])})(window,'partytown','forward');/* Partytown 0.4.5 - MIT builder.io */
    !function(t,e,n,i,r,o,a,d,s,c,p,l){function u(){l||(l=1,"/"==(a=(o.lib||"/~partytown/")+(o.debug?"debug/":""))[0]&&(s=e.querySelectorAll('script[type="text/partytown"]'),i!=t?i.dispatchEvent(new CustomEvent("pt1",{detail:t})):(d=setTimeout(w,1e4),e.addEventListener("pt0",f),r?h(1):n.serviceWorker?n.serviceWorker.register(a+(o.swPath||"partytown-sw.js"),{scope:a}).then((function(t){t.active?h():t.installing&&t.installing.addEventListener("statechange",(function(t){"activated"==t.target.state&&h()}))}),console.error):w())))}function h(t){c=e.createElement(t?"script":"iframe"),t||(c.setAttribute("style","display:block;width:0;height:0;border:0;visibility:hidden"),c.setAttribute("aria-hidden",!0)),c.src=a+"partytown-"+(t?"atomics.js?v=0.4.5":"sandbox-sw.html?"+Date.now()),e.body.appendChild(c)}function w(t,n){for(f(),t=0;t<s.length;t++)(n=e.createElement("script")).innerHTML=s[t].innerHTML,e.head.appendChild(n);c&&c.parentNode.removeChild(c)}function f(){clearTimeout(d)}o=t.partytown||{},i==t&&(o.forward||[]).map((function(e){p=t,e.split(".").map((function(e,n,i){p=p[i[n]]=n+1<i.length?"push"==i[n+1]?[]:p[i[n]]||{}:function(){(t._ptf=t._ptf||[]).push(i,arguments)}}))})),"complete"==e.readyState?u():(t.addEventListener("DOMContentLoaded",u),t.addEventListener("load",u))}(window,document,navigator,top,window.crossOriginIsolated);</script>
  </head>
  <body>
    <ol>
      <li>
        <a href="/">Top</a>
      </li>
      <li>
        <a href="/categories">Category</a>
      </li>
      <li>
        <a>Article1</a>
      </li>
    </ol>
  </body>
</html>
```

**Expected behavior**

Nothing happens, no errors.

**Partytown version**

`0.4.5`

**Additional context**

We are using `partytown` via `"@astrojs/partytown": "^0.1.4"` + `"astro": "^1.0.0-beta.35"`.

**Comment by steve8708** [comment]
hey @leader22 - we've just created an issue template, could you update your issue here to provide all the information the template asks for? it'll help us look into your issue https://github.com/BuilderIO/partytown/issues/new?assignees=&labels=&template=bug_report.md&title=

pasting below to make things easier:

---

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Reproduction link**
Please include a link to a Stackblitz or Codesandbox reproducing the issue. We will need to see the issue reproduced with hand-written code - we can't debug giant minified third party scripts directly. If you do not include a clean and simple reproduction of your issue, we won't be able to look into it until you do.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

**Comment by leader22** [comment]
Thanks @steve8708 , I updated. 😺

**Comment by steve8708** [comment]
Perfect, thanks @leader22!

**Comment by willpaige** [comment]
Any update on this, I'm getting a similar issue on my personal website https://will.paige.me.uk/ any links throw the error `TypeError: Failed to construct 'URL': Invalid URL`

**Comment by codeithuman** [comment]
We're experiencing this issue as well. And, of course, we're using an `<a>` tag to trigger JavaScript when we should technically be using a `<button>` tag. I believe switching our `<a>` tag to a `<button>` tag would fix the issue.

**Comment by gioboa** [comment]
Closed by #523
